packages/mui-joy/src/Sheet/Sheet.tsx

```diff
@@ -0,0 +1,131 @@
+import { unstable_composeClasses as composeClasses } from '@mui/base';
+import { OverridableComponent } from '@mui/types';
+import { unstable_capitalize as capitalize } from '@mui/utils';
+import clsx from 'clsx';
+import PropTypes from 'prop-types';
+import * as React from 'react';
+import { useThemeProps } from '../styles';
+import styled from '../styles/styled';
+import { getSheetUtilityClass } from './sheetClasses';
+import { SheetProps, SheetTypeMap } from './SheetProps';
+
+const useUtilityClasses = (ownerState: SheetProps) => {
+  const { elevation, variant, color } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      variant && `variant${capitalize(variant)}`,
+      color && `color${capitalize(color)}`,
+      elevation && `elevation${capitalize(elevation)}`,
+    ],
+  };
+
+  return composeClasses(slots, getSheetUtilityClass, {});
+};
+
+const SheetRoot = styled('div', {
+  name: 'MuiSheet',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [
+      styles.root,
+      styles[ownerState.variant],
+      ownerState.variant === 'elevation' && styles[`elevation${ownerState.elevation}`],
+    ];
+  },
```

Comment:
```suggestion
  overridesResolver: (props, styles) => styles.root,
```

Theming with non-slot keys is not possible anymore (starts in Joy). We will try to create a Codemod for Material in v6 https://github.com/mui/material-ui/issues/30412